### ISSUE TITLE
Improve Developer Install Docs to ensure CouchDB 2.3.x is installed

### ISSUE THREAD

**Issue by Ths2-9Y-LqJt6** [medic/cht-core#6381]
**What feature do you want to improve?**
The steps to set up the [developer environment](https://github.com/medic/cht-core/blob/master/DEVELOPMENT.md), specifically CouchDB installation.

**Describe the improvement you'd like**
Per the discussion on #6289, we should explicitly call out that the default steps for installing  CouchDB 2.3.x for Ubuntu >18.04 will result in installing CouchDB 3.0.0 instead. I'm not quite sure what this should be, but @garethbowen suggested a CouchDB Docker container to be more cross OS compatible.  What ever the end result, I suggest the main goal should be freeze the version we document the install steps for until we support the CouchDB 3 features/limitations per #6289.

**Describe alternatives you've considered**
One work around for Ubuntu is to use the [CouchDB snap](https://docs.couchdb.org/en/latest/install/snap.html) which I tested and works..  Another work around would be to [specify the version in apt](https://linoxide.com/linux-command/install-specific-version-package-apt-get/). Testing `sudo apt-get install couchdb=2.3.1~bionic -V` worked for me as well.

**Additional context**
None!

**Comment by garethbowen** [comment]
Thanks for raising this @Ths2-9Y-LqJt6 !

@SCdF If I recall correctly, you use CouchDB in Docker for development. Would you recommend this setup? Do you think we should change the instructions to the Docker instructions? I think it'd make it easier to get the right version, and also the instructions would be more cross platform compatible.

**Comment by SCdF** [comment]
@garethbowen yeah I like it, as it makes it easier to control versioning, can have multiple entire installs and I don't have to rely on OS packages so much.

The downside is the standard docker one: if you have an older non-linux computer you're running it in a VM and all the perf issues that this comes with. I think it's a better fool-proof default though.

PR created

**Comment by garethbowen** [comment]
Merged.

**Comment by Ths2-9Y-LqJt6** [comment]
@SCdF  - I know this PR is already merged and issue is closed, but I just noticed I [referenced](https://github.com/medic/cht-core/pull/6384#discussion_r417747000) `COUCH_NODE_NAME=nonode@nohost` in my testing, but we ended up merging `COUCH_NODE_NAME=noname@nohost` to master (note `noname` vs `nonode`).  

Re-re-testing the steps now, they don't work with `noname` but do with `nonode` for me. I'm hoping this has to do with my docker install being wonky and everyone else's uses `noname` so no fix is need.  On the off chance that is not the case, I wanted to call it out in case it is helpful.  Happy to open another issue or do what ever is helpful!
